src/jobs/jobs.service.ts

```diff
@@ -0,0 +1,55 @@
+import { HttpService } from '@nestjs/axios'
+import { Injectable } from '@nestjs/common'
+import { firstValueFrom } from 'rxjs'
+
+const API_BASE = 'https://api.ashbyhq.com/jobPosting'
+const ASHBEY_KEY = process.env.ASHBEY_KEY
+
+interface FetchJobsParams {
+  listedOnly?: boolean
+  jobPostingId?: string
+}
+
+@Injectable()
+export class JobsService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async findAll() {
+    try {
+      const jobs = await this.fetchJobs('list', { listedOnly: true })
+      return jobs || []
+    } catch (error) {
+      console.error('Error during fetch:', error.message)
+      return []
+    }
```

Comment:
I really think the API should bubble errors like this upward to result in something meaningful, not just an empty response.

For example.. if something's wrong w/ this server-to-server API call, and we just return `200` w/ an empty `[]` in the body, then the client will just render an empty page/list and we won't _know_ something's wrong in PROD until someone bothers to visit the jobs-listing page and then we have a production bug on our hands w/ a 🔥 we gotta put out.

It's typically better to have the mechanisms in place to be proactive, as opposed to reactive, to these types of errors.

My preference would be to log the error using the idiomatic NestJS pattern for the logger: https://konfigthis.com/sdk/#:~:text=Pricing,SDKs%20you%20use.

Then allow this error to be `throw`n up to the caller method in the controller, and then, based on the error, return a `502 Bad Gateway`, since our API is acting as a gateway for the Ashby service behind it.  (Similar to what you did here: https://github.com/thegoodparty/gp-api/pull/4/files#diff-e657b2661a51df80ffee31a7065014c030edfdf7534cb0ab6a0f29d2234aee99R17 albeit, not exactly the same)

This way, the _front-end_ app can then properly introspect the response code, and act accordingly (i.e. throw a client error that FullStory will pick up and let us know something's not right)